src/document/mod.rs

```diff
@@ -84,113 +116,241 @@ impl DocumentEncryptOpts {
+    /// The new contents will be encrypted, and which users and groups are granted access
+    /// will remain unchanged.
     ///
     /// # Arguments
-    /// - `id` - ID of document to update.
-    /// - `new_document_data` - Updated document content to encrypt.
+    /// - `id` - Unique ID of the document to update
+    /// - `new_document_data` - Updated bytes to encrypt
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// let new_data = "more secret data".as_bytes();
+    /// let encrypted = sdk.document_update_bytes(&document_id, new_data).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_update_bytes(
         &self,
         id: &DocumentId,
         new_document_data: &[u8],
     ) -> Result<DocumentEncryptResult>;
 
-    /// Decrypts the provided encrypted document and returns details about the document as well as its decrypted bytes.
+    /// Modifies or removes a document's name.
     ///
-    /// # Arguments
-    /// - `encrypted_document` - Bytes of encrypted document. Should be the same bytes returned from `document_encrypt`.
-    ///
-    /// # Returns
-    /// `Result<DocumentDecryptResult>` Includes metadata about the provided document as well as the decrypted document bytes.
-    async fn document_decrypt(&self, encrypted_document: &[u8]) -> Result<DocumentDecryptResult>;
-
-    /// Update a document name to a new value or clear its value.
+    /// Returns the updated metadata of the document.
     ///
     /// # Arguments
-    /// - `id` - ID of the document to update
-    /// - `name` - New name for the document. Provide a Some to update to a new name and a None to clear the name field.
+    /// - `id` - Unique ID of the document to update
+    /// - `name` - New name for the document. Provide a `Some` to update to a new name or a `None` to clear the name field.
     ///
-    /// # Returns
-    /// `Result<DocumentMetadataResult>` Metadata about the document that was updated.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// use std::convert::TryFrom;
+    /// let new_name = DocumentName::try_from("updated")?;
+    /// let document_meta = sdk.document_update_name(&document_id, Some(&new_name)).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_update_name(
         &self,
         id: &DocumentId,
         name: Option<&DocumentName>,
     ) -> Result<DocumentMetadataResult>;
 
-    /// Grant access to a document. Recipients of document access can be either users or groups.
+    /// Grants decryption access to a document for the provided users and/or groups.
```

Comment:
I wonder if we should talk more about partial success failure in the main comment block. Both here and for revoke.

Reply:
I think that the "Returns lists of successful and failed grants" implies the partial success, and the `Errors` section clarifies it